### PR TITLE
[FIX] stock_by_warehouse: display stock by warehouse in widget T#79454

### PR DESCRIPTION
**Pull request by xmglord**
There was an error when trying to load the stock by warehouse widget caused by the incorrectly use of `onWillUpdateProps` that was change in [1], this commit fixes that error and also adapt the stock by warehouse field to be computed. Finally, some styles were adapted to have a better display and to use recent version of bootstrap.

[1]: https://github.com/odoo/odoo/commit/d682871a

### CODE REVIEW

**Comment by xmglord** [comment]
@rolandojduartem @luisg123v please review.

File: stock_by_warehouse/tests/test_stock_available_unreserved.py
Comment (xmglord) [inline stock_by_warehouse/tests/test_stock_available_unreserved.py]: Even with the Json field the assertFalse didn't worked @luisg123v

**Review by luisg123v — APPROVED** [review]
LGTM :+1: 

I'll wait for @rolandojduartem's approval to merge.

File: stock_by_warehouse/tests/test_stock_available_unreserved.py
Comment (luisg123v) [reply stock_by_warehouse/tests/test_stock_available_unreserved.py]: I only want to ask: Why doesn't this happen in `stock_by_warehouse_sale`?

Comment (xmglord) [reply stock_by_warehouse/tests/test_stock_available_unreserved.py]: because it doesn't have test

File: stock_by_warehouse/models/product_product.py
Comment (rolandojduartem) [inline stock_by_warehouse/models/product_product.py]: is the else "{}" required after the change to json field?

File: stock_by_warehouse/static/src/components/warehouse/warehouse_field.esm.js
Comment (rolandojduartem) [inline stock_by_warehouse/static/src/components/warehouse/warehouse_field.esm.js:33]: is json.parse required after the change to json field?

**Comment by rolandojduartem** [comment]
Screenshot from 2024-05-03 10-28-30

Screenshot from 2024-05-03 10-28-22

Functionally it does not work :(

File: stock_by_warehouse/models/product_product.py
Comment (xmglord) [reply stock_by_warehouse/models/product_product.py]: No with False seems to be enough. I've update it

File: stock_by_warehouse/static/src/components/warehouse/warehouse_field.esm.js
Comment (xmglord) [reply stock_by_warehouse/static/src/components/warehouse/warehouse_field.esm.js:33]: in that line no but because is initializing but in the other it is necessary.

**Comment by xmglord** [comment]
> Screenshot from 2024-05-03 10-28-30
> 
> Screenshot from 2024-05-03 10-28-22
> 
> Functionally it does not work :(

For me it is working:
image
 :c

File: stock_by_warehouse/models/product_product.py
Comment (xmglord) [reply stock_by_warehouse/models/product_product.py]: Actually removing that and changing it to False solve the change in tests @luisg123v

File: stock_by_warehouse/static/src/components/warehouse/warehouse_field.esm.js
Comment (rolandojduartem) [reply stock_by_warehouse/static/src/components/warehouse/warehouse_field.esm.js:33]: if it is required, it would be required in the both cases or not required in the both cases, that is weird

Comment (xmglord) [reply stock_by_warehouse/static/src/components/warehouse/warehouse_field.esm.js:33]: Is weird because if I remove it from that line everything works fine but if removed from formatData then it fails.

**Comment by rolandojduartem** [comment]
Functionally it is OK. @xmglord it did not worked because you had renamed the js file, but you did not changed the manifest, so it was loading nothing. It is Ok functionally

File: stock_by_warehouse/static/src/components/warehouse/warehouse_field.esm.js
Comment (rolandojduartem) [reply stock_by_warehouse/static/src/components/warehouse/warehouse_field.esm.js:33]: OK, NVM, @xmglord, it is required in the both lines, js gets the value as string, the Json.parse converts that to a json object

**Comment by rolandojduartem** [comment]
Just one comment more, the spanish translation was not updated during the migration process. @xmglord 

cc @luisg123v 

PD. After that, @luisg123v could merge this PR whenever he wants, I think it LGTM

**Comment by xmglord** [comment]
@luisg123v already update the translation file so this is ready to merge.

cc: @rolandojduartem

**Pull request merged**
